File: pbnorm.py
#!/usr/bin/env python3

import os
import re

def pbnorm():
    pb = re.compile(r"^PB_Kontoauszug_KtoNr_\d{10}_(\d{2})-(\d{2})-(\d{4})_\d{6}.pdf$")
    files = [f for f in os.listdir(".") if os.path.isfile(f)]
    for f in files:
        match = pb.match(f)
        if match:
            (day, month, year) = (match.group(1), match.group(2), match.group(3))
            to = "Postbank-{}-{}-{}.pdf".format(year, month, day)
            os.rename(f, to)

if __name__ == "__main__":
    pbnorm()

File: pbnorm.ts
#!/usr/bin/env -S deno run --allow-read --allow-write

/*
"PB_Kontoauszug_KtoNr_1234567890_06-01-2020_043030.pdf" \
"PB_Kontoauszug_KtoNr_1111111111_06-02-2020_043030.pdf" \
"PB_Kontoauszug_KtoNr_2222222222_06-03-2020_043030.pdf" \
"PB_Kontoauszug_KtoNr_3333333333_06-04-2020_043030.pdf" \
"PB_Kontoauszug_KtoNr_4444444444_06-05-2020_043030.pdf"
*/

const re = /^PB_Kontoauszug_KtoNr_\d{10}_(\d{2})-(\d{2})-(\d{4})_\d{6}.pdf$/;

for (const entry of Deno.readDirSync(".")) {
    if (entry.isFile) {
        const match = entry.name.match(re);
        if (match) {
            const [_, day, month, year] = match;
            const to = `Postbank-${year}-${month}-${day}.pdf`
            Deno.renameSync(entry.name, to);
        }
    }
}